src/Cognito.re

```diff
@@ -83,4 +165,50 @@ let signUp =
         });
       };
     });
+};
+
+open SignInHelper;
+external makeSignInResponse: 't => signInResponseDecoder = "%identity";
+external toJsDict: 'a => Js.Dict.t(Js.Json.t) = "%identity";
+let signIn = (~username: string, ~password: string, ()) => {
+  let signInParams = authenticationParameters(~username, ~password)->toJsDict;
+  /* create and object of this shape using our `authenticationParameters` type
+     ```json
+     {
+       AuthParameters : {
+         "USERNAME": "idkjs",
+         "PASSWORD": "somepassword",
+       }
+     }
+     ```
+     At this point we know we have a valid js object which can test with `Js.Json.test(signInParams, Object)`
+     Since we know its valid we will change its type to Js.Dict.t(Js.Json.t) which is expected by out params dict. We then use `Js.Dict.set(params, "AuthParameters", Js.Json.object_(signInParams));`to push it to the `AuthParameters` key expected by the cognito client.
+      */
+  let params = Js.Dict.empty();
+  Js.Dict.set(params, "AuthParameters", Js.Json.object_(signInParams));
+  Client.request(`InitiateAuth, params)
```

Comment:
I like the variant approach here for the operation!

Reply:
Nice. Can implement with auth flow type or convert to the to string method you used in your latest commit. I think `toString` is native compliant so we should go with that. Which do you prefer? 

ToString I am referring to:
https://github.com/pckilgore/reason-cognito/blob/f1b437a8eaafd19abafae9443069537bbce03ec0/src/Types.re#L20-L33

Reply:
I'm not sure what you're talking about but I think I'm about to learn something good so go for it.

Reply:
`bucklescript` isnt supported in native just yet. So we do `bs.whatever` it wont work on native, like this:
```reason
[@bs.deriving jsConverter]
type operation = [
  | [@bs.as "SignUp"] `SignUp
  | [@bs.as "SignIn"] `SignIn
  | [@bs.as "ConfirmSignUp"] `ConfirmSignUp
  | [@bs.as "ChangePassword"] `ChangePassword
  | [@bs.as "RespondToAuthChallenge"] `RespondToAuthChallenge
  | [@bs.as "ForgotPassword"] `ForgotPassword
  | [@bs.as "ConfirmForgotPassword"] `ConfirmForgotPassword
  | [@bs.as "InitiateAuth"] `InitiateAuth
];
```

Where as taking the extra set write a `operataionToString` function, will compile in native.

Reply:
Ahh I see.  Yeah go for it.  The `toString` way is a bit more boilerplate than the bs helper, but it's super easy to refactor either way so go ahead and do what feels right.